libfs/fs.go

```diff
@@ -547,10 +547,15 @@ func (fs *FS) readDir(n libkbfs.Node) (fis []os.FileInfo, err error) {
 
 	fis = make([]os.FileInfo, 0, len(children))
 	for name, ei := range children {
+		child, _, err := fs.config.KBFSOps().Lookup(fs.ctx, n, name)
```

Comment:
This means that we're now explicitly doing an on-demand lookup for any `readDir`. It'll deepen the prefetch tree, which may or may not be what we want. I'm not necessarily expressing an opinion on whether this is correct, I just wanted to make sure it was considered.

Reply:
I did consider that and frowned as I did so.  But I don't see a better way to get the proper permissions for all the children.  Basically, the billy interface diverges a bit from the normal FUSE interface in that it asks for the permissions up-front on a readdir, wherease FUSE does not.  But most of the time, the OS will end up doing a stat on all the children anyway, so I think in the end it works out to be the same thing.

It does mean some extra prefetching for some go-git use cases that might not have happened before.  I don't think `ReadDir` is used extensively, so I think it's ok.

Reply:
You could pass the request priority into the `Lookup`, to prevent it from triggering prefetches.

Reply:
Yeah, that's an option if it becomes an issue.  I don't love exposing prefetching options outside of the libkbfs package, so I vote for tackling this later if it sucks.

Reply:
Sounds good to me!